### PR TITLE
Remaining reviews on my votes buttons

### PR DESCRIPTION
**Pull request by fcamblor**
Displaying remaining reviews on my votes screen (+ trophy badge when `remainingReviews = 0`), as suggested by @aheritier on #206

All_my_votes_-_Devoxx_France_2020

Important : this PR requires #206 (that's why I put my commit as an extension of #206)

### CODE REVIEW

File: app/views/CFPAdmin/allMyVotes.scala.html [hunk 1]
```diff
@@ -14,20 +14,38 @@ <h3><i class="fas fa-flask"></i> Your personnal leaderboard</h3>
                     <div style="margin: 3px 0px">
                     <i class="fas fa-filter"></i> @Messages("ar.proposalType") :
                     @models.ConferenceDescriptor.ConferenceProposalTypes.ALL.map{confType=>
-                        <a href="@routes.CFPAdmin.allMyVotes(confType.id, selectedTrack)" class="btn btn-sm @if(confType.id==talkType) { btn-success } else { btn-primary }">@Messages(confType.id)</a>
+                        <a href="@routes.CFPAdmin.allMyVotes(confType.id, selectedTrack)" class="btn btn-sm @if(confType.id==talkType) { btn-success } else { btn-primary }">
+                            @if(proposalNotReviewedCountByType.get(confType.id).getOrElse(0) == 0){
+                                <i class="fas fa-trophy"></i>
+                            }
+                            @Messages(confType.id)
+                            (@{proposalNotReviewedCountByType.get(confType.id).getOrElse(0) })
```
Comment (aheritier) [inline app/views/CFPAdmin/allMyVotes.scala.html]:
```suggestion
                            <span class="badge badge-pill badge-light">@{proposalNotReviewedCountByType.get(confType.id).getOrElse(0) }</span>
```

File: app/views/CFPAdmin/allMyVotes.scala.html [hunk 2]
```diff
@@ -14,20 +14,38 @@ <h3><i class="fas fa-flask"></i> Your personnal leaderboard</h3>
...
                     <br/>
                     <div style="margin: 3px 0px">
                         <i class="fas fa-filter"></i> @Messages("cfp.filter.on") :
-                        <a href="@routes.CFPAdmin.allMyVotes(talkType, None)" class="btn btn-sm @if(selectedTrack==None) { btn-success } else { btn-primary }">All tracks</a>
+                        <a href="@routes.CFPAdmin.allMyVotes(talkType, None)" class="btn btn-sm @if(selectedTrack==None) { btn-success } else { btn-primary }">
+                            @if(proposalNotReviewedCountByType.get(talkType).getOrElse(0) == 0) {
+                                <i class="fas fa-trophy"></i>
+                            }
+                            All tracks
+                            (@{proposalNotReviewedCountByType.get(talkType).getOrElse(0) })
```
Comment (aheritier) [inline app/views/CFPAdmin/allMyVotes.scala.html]:
```suggestion
                            <span class="badge badge-pill badge-light">@{proposalNotReviewedCountByType.get(talkType).getOrElse(0) }</span>
```

File: app/views/CFPAdmin/allMyVotes.scala.html [hunk 3]
```diff
@@ -14,20 +14,38 @@ <h3><i class="fas fa-flask"></i> Your personnal leaderboard</h3>
...
+                        </a>
                         @models.ConferenceDescriptor.ConferenceTracks.ALL.map{track =>
-                            <a href="@routes.CFPAdmin.allMyVotes(talkType, Some(track.id))" class="btn btn-sm @if(selectedTrack==Some(track.id)) { btn-success } else { btn-primary }">@Messages(track.label)</a>
+                            <a href="@routes.CFPAdmin.allMyVotes(talkType, Some(track.id))" class="btn btn-sm @if(selectedTrack==Some(track.id)) { btn-success } else { btn-primary }">
+                                @if(proposalNotReviewedCountForCurrentTypeByTrack.get(track.id).getOrElse(0) == 0){
+                                    <i class="fas fa-trophy"></i>
+                                }
+                                @Messages(track.label)
+                                (@{proposalNotReviewedCountForCurrentTypeByTrack.get(track.id).getOrElse(0) })
```
Comment (aheritier) [inline app/views/CFPAdmin/allMyVotes.scala.html]:
```suggestion
                                <span class="badge badge-pill badge-light">@{proposalNotReviewedCountForCurrentTypeByTrack.get(track.id).getOrElse(0) }</span>
```

File: app/views/GoldenTicketController/allMyGoldenTicketVotes.scala.html [hunk 4]
```diff
@@ -16,20 +16,38 @@ <h3><i class="fas fa-flask"></i> @Messages("gt.personnal")</h3>
                     <div style="3px 0px">
                     <i class="fas fa-filter"></i> @Messages("ar.proposalType") :
                     @models.ConferenceDescriptor.ConferenceProposalTypes.ALL.filter(models.ConferenceDescriptor.ConferenceProposalConfigurations.accessibleTypeToGoldenTicketReviews(_)).map{confType=>
-                        <a href="@routes.GoldenTicketController.allMyGoldenTicketVotes(confType.id, selectedTrack)" class="btn btn-sm @if(confType.id==talkType) { btn-success } else { btn-primary }">@Messages(confType.id)</a>
+                        <a href="@routes.GoldenTicketController.allMyGoldenTicketVotes(confType.id, selectedTrack)" class="btn btn-sm @if(confType.id==talkType) { btn-success } else { btn-primary }">
+                            @if(proposalNotReviewedCountByType.get(confType.id).getOrElse(0) == 0){
+                                <i class="fas fa-trophy"></i>
+                            }
+                            @Messages(confType.id)
+                            (@{proposalNotReviewedCountByType.get(confType.id).getOrElse(0) })
```
Comment (aheritier) [inline app/views/GoldenTicketController/allMyGoldenTicketVotes.scala.html]:
```suggestion
                            <span class="badge badge-pill badge-light">@{proposalNotReviewedCountByType.get(confType.id).getOrElse(0) }</span>
```

File: app/views/GoldenTicketController/allMyGoldenTicketVotes.scala.html [hunk 5]
```diff
@@ -16,20 +16,38 @@ <h3><i class="fas fa-flask"></i> @Messages("gt.personnal")</h3>
...
                     <br/>
                     <div style="3px 0px">
                         <i class="fas fa-filter"></i> @Messages("cfp.filter.on") :
-                        <a href="@routes.GoldenTicketController.allMyGoldenTicketVotes(talkType, None)" class="btn btn-sm @if(selectedTrack==None) { btn-success } else { btn-primary }">All tracks</a>
+                        <a href="@routes.GoldenTicketController.allMyGoldenTicketVotes(talkType, None)" class="btn btn-sm @if(selectedTrack==None) { btn-success } else { btn-primary }">
+                            @if(proposalNotReviewedCountByType.get(talkType).getOrElse(0) == 0) {
+                                <i class="fas fa-trophy"></i>
+                            }
+                            All tracks
+                            (@{proposalNotReviewedCountByType.get(talkType).getOrElse(0) })
```
Comment (aheritier) [inline app/views/GoldenTicketController/allMyGoldenTicketVotes.scala.html]:
```suggestion
                            <span class="badge badge-pill badge-light">@{proposalNotReviewedCountByType.get(talkType).getOrElse(0) }</span>
```

File: app/views/GoldenTicketController/allMyGoldenTicketVotes.scala.html [hunk 6]
```diff
@@ -16,20 +16,38 @@ <h3><i class="fas fa-flask"></i> @Messages("gt.personnal")</h3>
...
+                        </a>
                         @models.ConferenceDescriptor.ConferenceTracks.ALL.map{track =>
-                            <a href="@routes.GoldenTicketController.allMyGoldenTicketVotes(talkType, Some(track.id))" class="btn btn-sm @if(selectedTrack==Some(track.id)) { btn-success } else { btn-primary }">@Messages(track.label)</a>
+                            <a href="@routes.GoldenTicketController.allMyGoldenTicketVotes(talkType, Some(track.id))" class="btn btn-sm @if(selectedTrack==Some(track.id)) { btn-success } else { btn-primary }">
+                                @if(proposalNotReviewedCountForCurrentTypeByTrack.get(track.id).getOrElse(0) == 0){
+                                    <i class="fas fa-trophy"></i>
+                                }
+                                @Messages(track.label)
+                                (@{proposalNotReviewedCountForCurrentTypeByTrack.get(track.id).getOrElse(0) })
```
Comment (aheritier) [inline app/views/GoldenTicketController/allMyGoldenTicketVotes.scala.html]:
```suggestion
                               <span class="badge badge-pill badge-light">@{proposalNotReviewedCountForCurrentTypeByTrack.get(track.id).getOrElse(0) }</span>
```

**Review by aheritier — CHANGES_REQUESTED** [review]
For a better readability I would propose to use real bootstrap badges @fcamblor 

Screenshot 2020-01-13 15 10 36

But I have a bug with real data. Counters aren't the number of reviews to do but the total of proposals

**Comment by fcamblor** [comment]
💯 @aheritier, just pushed the fix 👍

**Comment by aheritier** [comment]
@fcamblor the fix for badges, but you fixed the issue with the counter values ?

**Comment by fcamblor** [comment]
> But I have a bug with real data. Counters aren't the number of reviews to do but the total of proposals

Didn't read your message until the end

I think the count is ok at the "Proposal type" level, but not at the "Track" level.
Will look at it

**Comment by aheritier** [comment]
@fcamblor it's strange because if you look at my screenshot you see that I have 517 conferences to review which is wrong even at the type level !!!!
I have 147 to review (screenshot from prod)

Screenshot 2020-01-13 16 05 10

**Comment by fcamblor** [comment]
Yes I found the problem : I'm not taking the track filter into consideration when displaying those numbers

**Comment by fcamblor** [comment]
force pushed the fix (twice), should be better now :-)

**Comment by fcamblor** [comment]
Added a commit fixing some ":" duplicates here and there in filtering labels

**Comment by aheritier** [comment]
I am testing 

```
* 72f21a95 (HEAD -> pr/207, origin/pr/207, fcamblor/badges-on-my-votes) fixed typo in type/track filtering label (':' displayed twice)
```

But it seems wrong.

In production for universities I have 

> Out of 37 talks, you have reviewed 34 talks and 3 reviews remain.

Screenshot 2020-01-13 18 22 32

But with the PR

> Out of 71 talks, you have reviewed 34 talks and 37 reviews remain.

Screenshot 2020-01-13 18 21 31

Production is right. We received 37 universities, I reviewed 34 and thus I have 3 remaining

In the PR the total (71 = 37 + my 34 reviews :-/ ) instead of 37
the number of reviews (37 = 3 +34 ) instead of 3

**Comment by fcamblor** [comment]
mmm are you sure about your redis dump/sync with production ?

Not saying that there isn't any bug, but I don't have the same behaviour locally (overall total of reviews seem ok here)

Cannot it be due to a "double import" of proposals (or something like this) ?

Redis commands to check (prefix it with `docker exec -ti cfp-devoxx_redis` if running with docker) :
- To list all proposals keys (check if there isn't any duplicate here for some universities) :
```
redis-cli hkeys Proposals
```

- To list entries for a given key :
```
redis-cli hget Proposals HGE-1930
```

Not being a Redis expert but :
- I think that HASHes should not have the same key twice (I don't expect the first command to return duplicates)
- For a given key, you're not supposed to have multiple values (second command), so I don't see how it could happen that you have multiple entries here

I'll triple check once at home but honestly I don't understand what's happening here :(

**Comment by aheritier** [comment]
if I run `hkeys Proposals` I get 6117 entries thus probably all the keys used for several years cc @nicmarti 
`hget Proposals HGE-1930` gives me one entry

**Comment by fcamblor** [comment]
my bad I think I got it

**Comment by fcamblor** [comment]
@aheritier can you try to test again ?

Ugly copy/paste spotted

(I didn't spotted it because locally, I voted exactly on the same talks on my GT & comitee accounts... pfff)

**Review by aheritier — APPROVED** [review]
it's all good.
it's beautiful and ok to merge

Screenshot 2020-01-13 20 04 47

cc @nicmarti

**Comment by aheritier** [comment]
@nicmarti merging this Pr will solve #206 too cc @fcamblor

**Comment by aheritier** [comment]
@nicmarti can you merge/deploy it please ?

**Comment by nicmarti** [comment]
Thanks for the PR @fcamblor. It should be deployed on clever cloud at 10h55 today

**Pull request merged**